src/fastoad/io/variable_io.py

```diff
@@ -49,6 +50,10 @@ def read(self, only: List[str] = None, ignore: List[str] = None) -> VariableList
         :param ignore: List of variable names that should be ignored when reading.
         :return: an VariableList instance where outputs have been defined using provided source
         """
+        if isinstance(self.data_source, str) and not isfile(self.data_source):
+            raise FileNotFoundError(
+                f'File "{self.data_source}" is unavailable for reading.'
+            ) from FastError()
```

Comment:
In some other files, when a line corrects an issue it is sometimes mentionned. Out of curiosity what's the "good practice" that decides when to fo it or not

Reply:
Well, I answered just above :)
Yet, do not consider it as absolute truth. That is also my own interpretation.